ssh/tailssh/tailssh.go

```diff
@@ -109,34 +109,87 @@ func (srv *server) OnPolicyChange() {
 	}
 }
 
-func (srv *server) newSSHServer() (*ssh.Server, error) {
-	ss := &ssh.Server{
+// conn represents a single SSH connection and it's associated
+// ssh.Server.
+type conn struct {
+	*ssh.Server
+	srv *server
+	ci  *sshConnInfo // set by NoClientAuthCallback
+
+	insecureSkipTailscaleAuth bool // used by tests.
+}
+
+func (c *conn) logf(format string, args ...any) {
+	if c.ci == nil {
+		c.srv.logf(format, args...)
+		return
+	}
+	format = fmt.Sprintf("%v: %v", c.ci.String(), format)
+	c.srv.logf(format, args...)
+}
+
+func (c *conn) PublicKeyHandler(ctx ssh.Context, key ssh.PublicKey) bool {
```

Comment:
// PublicKeyHandler is the gossh.Server.PublicKeyHandler implementation.

or whatever